float64[9] basis
float64[3] direction
float64[3] next_joint
float64[3] prev_joint
float64[3] center
bool is_valid
float64 length
float64 width
